### PR TITLE
call cancel function only on ticker stop

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
otherwise it will cancel other go routines of the syncer

### CODE REVIEW

File: pkg/dbsyncers/policy_syncer.go [hunk 1]
```diff
@@ -46,16 +46,25 @@ func (syncer *policyDBSyncer) Start(stopChannel <-chan struct{}) error {
...
+			if cancelFunc != nil {
+				cancelFunc()
+			}
+
 			return
 
 		case <-ticker.C:
-			ctxWithTimeout, cancelFunc := context.WithTimeout(ctx, syncer.syncInterval*timeoutInIntervalPeriods)
+			ctxWithTimeout, cancelFunc = context.WithTimeout(ctx, syncer.syncInterval*timeoutInIntervalPeriods)
 			syncer.sync(ctxWithTimeout)
```
Comment (nirrozenbaum) [inline pkg/dbsyncers/policy_syncer.go:57]:
```
ctxWithTimeout, _ = context.WithTimeout(ctx, syncer.syncInterval*timeoutInIntervalPeriods)
syncer.sync(ctxWithTimeout)
```

File: pkg/dbsyncers/policy_syncer.go [hunk 2]
```diff
@@ -46,16 +46,25 @@ func (syncer *policyDBSyncer) Start(stopChannel <-chan struct{}) error {
...
+		cancelFunc     context.CancelFunc
+		ctxWithTimeout context.Context
+	)
+
 	for {
 		select {
 		case <-ctx.Done(): // we have received a signal to stop
 			ticker.Stop()
+
+			if cancelFunc != nil {
```
Comment (nirrozenbaum) [inline pkg/dbsyncers/policy_syncer.go]: this is not needed

File: pkg/dbsyncers/policy_syncer.go [hunk 3]
```diff
@@ -46,16 +46,25 @@ func (syncer *policyDBSyncer) Start(stopChannel <-chan struct{}) error {
 func (syncer *policyDBSyncer) periodicSync(ctx context.Context) {
 	ticker := time.NewTicker(syncer.syncInterval)
 
+	var (
```
Comment (nirrozenbaum) [inline pkg/dbsyncers/policy_syncer.go]: this is not needed

**Review by nirrozenbaum — COMMENTED** [review]
some of the lines are not needed.
cancelling the parent context when getting stop message will cancel child context as well.

**Comment by vadimeisenbergibm** [comment]
@nirrozenbaum 
right, fixed your comments in https://github.com/open-cluster-management/hub-of-hubs-status-sync/pull/4/commits/6088b3e0624b01299a14274ccf944b5a2d1c6493

**Review by nirrozenbaum — APPROVED** [review]

**Pull request merged**